numpy/core/fromnumeric.py

```diff
@@ -63,10 +82,46 @@ def _wrapfunc(obj, method, *args, **kwds):
     # in its class, but its signature is not identical to that
     # of NumPy's. This situation has occurred in the case of
     # a downstream library like 'pandas'.
-    except (AttributeError, TypeError):
+    except (AttributeError, TypeError) as e:
```

Comment:
Could except the errors separately, directly check the message instead of calling `_is_compat_error`, and unindent the `return _wrapit(obj, method, *args, **kwds)`. A little code duplication doesn't hurt and it is easier to follow than looking up another function.